packages/flutter_tools/lib/src/base/net.dart

```diff
@@ -120,7 +121,7 @@ class Net {
           exitCode: kNetworkProblemExitCode,);
       }
       _logger.printError(error.toString());
-      rethrow;
+      return error;
```

Comment:
Maybe these should all rethrow, `fetchUrl` can do a `try {} catch on IOException {}` with a `continue` in the `catch` if there are still attempts left? `doesRemoteFileExist` would also need a `try {} catch {}`, too, then I guess.